### DISCUSSION TITLE
How does the community detection algorithm Louvain handle directed weighted graphs in Gephi

### DISCUSSION THREAD

**Discussion by daibitao19** [gephi/gephi#2329] [Q&A]
Hi，dear author.
I have some questions about whether the community detection algorithm in Gephi can handle directed weighted graphs.
I looked up a paper on Fast winnings of communities in large networks and it did not seem to deal with directed graphs.I would like to know how Gephi handles the weight of two nodes joining edges in a directed graph for community detection.For example, if node 3 to node 4 has an edge and its weight is 20, node 4 to node 3 has an edge and its weight is 30, then what is the definition formula of modularity when selecting community detection in Gephi?What is the connection weight of node 3 and node 4?Is the community detection algorithm in Gephi only capable of processing undirected weighted graphs? Even if the directed weighted graphs are selected, they are treated as undirected weighted graphs.The weight of the connection between node 3 and node 4 is 20+30=50, right?

Best wishes
Hope for your letter 
                                                      William

**Comment by EG-CHEN** [comment]
Hi!
I have the same question with you. i think that LOUVAIN can handle directed weighted graphs. i use  two same graph(one is directed and the other is undirected) to test the algorithm, , the result is different.  In the data input process, you need to choose the graph is directed or undirected,  the soft can recognise it and produce different result. BUT i DON'T know what is aglorithm formula. Hope it is helpful for you！

**Comment by daibitao19** [comment]
您好，我已经收到您的邮件，稍后我将尽快给你回复。

**Comment by daibitao19** [comment]
Thanks，indeed，I conducted the action like what you just said. I just wanted to，acquire the formulation of Louvain in Gephi. But the reference listed in Gephi&nbsp; only shows the unweighted Louvain. Anyway， thanks for your kindness.



---Original---
From: ***@***.***&gt;
Date: Wed, Mar 16, 2022 15:05 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [gephi/gephi] How does the community detection algorithm Louvain handle directed weighted graphs in Gephi (Discussion #2329)




 
Hi!
 I have the same question with you. i think that LOUVAIN can handle directed weighted graphs. i use  two same graph(one is directed and the other is undirected) to test the algorithm, , the result is different.  In the data input process, you need to choose the graph is directed or undirected,  the soft can recognise it and produce different result. BUT i DON'T know what is aglorithm formula. Hope it is helpful for you！
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Reply by EG-CHEN** [reply]
我也是想找公式，但现在还没找到。我现在找到一个链接（https://blog.csdn.net/panbaoran913/article/details/118369265）

**Comment by daibitao19** [comment]
已经404了



---Original---
From: ***@***.***&gt;
Date: Wed, Mar 16, 2022 15:18 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [gephi/gephi] How does the community detection algorithm Louvainhandle directed weighted graphs in Gephi (Discussion #2329)




 
我也是想找公式，但现在还没找到。我现在找到一个链接（https://blog.csdn.net/panbaoran913/article/details/118369265）
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by EG-CHEN** [comment]
https://blog.csdn.net/panbaoran913/article/details/118369265

**Comment by daibitao19** [comment]
嗯嗯，我当时论文没自己编程，直接gephi算了



---Original---
From: ***@***.***&gt;
Date: Wed, Mar 16, 2022 16:04 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [gephi/gephi] How does the community detection algorithm Louvainhandle directed weighted graphs in Gephi (Discussion #2329)




 
https://blog.csdn.net/panbaoran913/article/details/118369265
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by hujiaying1999** [comment]
I also have the same question, but I tried many ways and got no answers....Actually, it is a very serious problem that people don't know what kind of formulation the tool use but still throw data to the tool.... I asked this question via twitter, but untill now, no replies...

**Reply by MHT123321** [reply]
Excuse me?has this problem been solved in the end?

**Comment by daibitao19** [comment]
您好，我已经收到您的邮件，稍后我将尽快给你回复。

**Comment by MHT123321** [comment]
Hello,Dear William,
I saw the question you raised in 2021. I want to know whether the community detection algorithm is processed according to the unoriented graph by default when processing the directed weighted graph in Gephi, and whether the weights of the A-B side and the B-A side are added?
Best wishes
Looking forward to your reply!
HH

**Reply by MHT123321** [reply]
@daibitao19

**Comment by EG-CHEN** [comment]
No



---Original---
From: ***@***.***&gt;
Date: Thu, May 8, 2025 09:49 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [gephi/gephi] How does the community detection algorithm Louvainhandle directed weighted graphs in Gephi (Discussion #2329)




 
Excuse me?has this problem been solved in the end?
